sonarts-core/src/rules/noNestedTemplateLiteralsRule.ts

```diff
@@ -45,12 +45,19 @@ export class Rule extends tslint.Rules.AbstractRule {
 
 class Visitor extends SonarRuleVisitor {
   public visitTemplateExpression(node: ts.TemplateExpression) {
-    const nestedTemplateLiteral = descendants(node).find(
-      child => is(child, ts.SyntaxKind.TemplateExpression) || is(child, ts.SyntaxKind.NoSubstitutionTemplateLiteral),
-    );
-    if (nestedTemplateLiteral) {
-      this.addIssue(nestedTemplateLiteral, Rule.MESSAGE);
-    }
+    this.raiseIssueIfNestedTemplate(node);
     super.visitTemplateExpression(node);
   }
+
+  public visitNoSubstitionTemplateLiteral(node: ts.NoSubstitutionTemplateLiteral) {
+    this.raiseIssueIfNestedTemplate(node);
+    super.visitNoSubstitionTemplateLiteral(node);
+  }
+
+  private raiseIssueIfNestedTemplate(node: ts.TemplateLiteral) {
```

Comment:
@andrea-guarino-sonarsource we usually use `checkXXX` as name for such functions